lib/util/git/index.ts

```diff
@@ -190,9 +189,11 @@ export async function validateGitVersion(): Promise<boolean> {
   return true;
 }
 
-async function fetchBranchCommits(): Promise<void> {
+async function fetchBranchCommits(preferUpstream = true): Promise<void> {
   config.branchCommits = {};
-  const opts = ['ls-remote', '--heads', config.url];
+  const url =
+    preferUpstream && config.upstreamUrl ? config.upstreamUrl : config.url;
```

Comment:
Coverage test fails on this line. Need to trigger `syncGit()` with preferUpstream==false in some test..